oteapi/datacache/datacache.py

```diff
@@ -109,10 +109,11 @@ def __init__(
             cache_dir = self.config.cacheDir
         if isinstance(cache_dir, str):
             cache_dir = Path(cache_dir)
-        if cache_dir.is_absolute():
+        if cache_dir.is_absolute():  # type: ignore[union-attr]
```

Comment:
I think for that specific use case I would make `DataCacheConfig` a subclass of `GenericConfig` - but yes - otherwise it seems you've caught my meaning.
And you should definitely keep the existing `GenericConfig`, but move all the methods you've made to a new class called `AttrDict` or similar. Also - there is actually no need to create a fancy class, since a simple pydantic model will work the same - in a sense. If you simply use the internal class `Config` to set `extra = "allow"` (see the [docs here](https://pydantic-docs.helpmanual.io/usage/model_config/)). Then one can instantiate the `configuration` parameter using a dictionary and always retrieve the keys either as an attribute or via the built-in `getattr()` function, where a default value can be included.